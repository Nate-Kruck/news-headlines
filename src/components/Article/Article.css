body {
  font-family: 'Droid Serif', serif;
  background-color: #f9f7f1;
  color: #2f2f2f;
  font-size: 14px;
}


.content {
  width: 100%;
  font-size: 0;
  line-height: 0;
  word-spacing: -.31em;
  display: inline-block;
  margin: 30px 2% 0 2%;
}

.time {
  text-transform: uppercase;
  border-bottom: 2px solid #2f2f2f;
  border-top: 2px solid #2f2f2f;
  padding: 5px 0 5px 0;
  text-align: center;
  position: relative;
}

.columns {
  font-size: 14px;
  line-height: 20px;
  list-style: none;
  width: 29.5%;
  display: inline-block;
  padding: 0 1% 0 1%;
  vertical-align: top;
  margin-bottom: 50px;
  transition: all .7s;
  border-left: 1px solid #2f2f2f;
}

.columns .title {
  text-align: center;
  line-height: normal;
  font-family: 'Playfair Display', serif;
  margin: 0 auto;
  font-weight: 400;
  font-style: italic;
  font-size: 15px;
  box-sizing: border-box;
  padding: 10px 0 10px 0;
}

.columns .author:before {
  border-top: 1px solid #2f2f2f;
  content: '';
  width: 100px;
  height: 7px;
  margin: 0 auto;
}

.columns .author:after {
  border-top: 1px solid #2f2f2f;
  content: '';
  width: 100px;
  height: 10px;
  margin: 0 auto;
}

.Article img {
  border-radius: 3px;
  -webkit-filter: sepia(80%) contrast(1) opacity(0.8);
  filter: sepia(80%) grayscale(1) contrast(1) opacity(0.8);
  mix-blend-mode: multiply;
  width: 100%;
}

.author {
  text-align: center;
  font-weight: 700;
  font-size: 14px;
}

.title h1 {
  color: black;
  text-transform: uppercase;
  border-bottom: 2px solid #2f2f2f;
  border-top: 2px solid #2f2f2f;
  padding: 12px 0 12px 0;
}

.columns li {
  border: 2px solid black;
  margin: 0 auto;
  text-align: center;
}