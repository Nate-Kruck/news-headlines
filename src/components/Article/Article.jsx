import React from 'react';
import PropTypes from 'prop-types';
import styles from './Article.css';

/* Line:15 I added target="_blank" rel="norferrer" 
  the later has to be added for security purposes */
function Article({ article }) {
  return (
    <div className={styles.mainGrid}>
      <li className={styles.Article}>
        <span className={styles.title}>
          <h1 data-testid="article">{article.title}</h1>
        </span>
        <span className={styles.imageContainer}>
          <a href={article.url} target="_blank" rel="noreferrer">
            <img src={article.imageUrl} alt={article.title} />
          </a>
        </span>
        <p className={styles.author}>{article.author}</p>
        <p className={styles.description}>{article.description}</p>
      </li>
    </div>
  );
}

Article.propTypes = {
  article: PropTypes.shape({
    title: PropTypes.string,
    author: PropTypes.string,
    description: PropTypes.string,
    imageUrl: PropTypes.string,
    url: PropTypes.string
  }).isRequired
};

export default Article;
