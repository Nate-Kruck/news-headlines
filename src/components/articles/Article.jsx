import React from 'react';
import PropTypes from 'prop-types';
import styles from '../articles/Articles.css';

function Article({ article }) {
  return (
    <div className={styles.Article}>
      <h1 data-testid="article">{article.title}</h1>
      <p>{article.author}</p>
      <p>{article.content}</p>
    </div>
  );
}

Article.propTypes = {
  article: PropTypes.shape({
    title: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
  }).isRequired
};

export default Article;
