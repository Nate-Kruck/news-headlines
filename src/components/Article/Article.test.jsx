import React from 'react';
import { 
  render,
  cleanup,
  screen
} from '@testing-library/react';
import Article from './Article';

describe('Article Testing', () => {
  afterEach(() => cleanup());
  
  it('render an Article', () => {
    const article = {
      title: 'some text',
      author: 'Bertholomule',
      description: 'Hey friends',
      imageUrl: 'http://placeholder.com/articles',
      url: 'nate-kruck.com'
    };
    const { asFragment } = render(<Article article={article} />);

    screen.getByText('some text');

    expect(asFragment()).toMatchSnapshot();
  });
});
