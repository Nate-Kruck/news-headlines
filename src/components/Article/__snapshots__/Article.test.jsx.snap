// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Testing render an Article 1`] = `
<DocumentFragment>
  <div
    class="content"
  >
    <div
      class="columns"
    >
      <li
        class="Article"
      >
        <span
          class="title"
        >
          <h1
            data-testid="article"
          >
            some text
          </h1>
        </span>
        <span
          class="imageContainer"
        >
          <a
            href="nate-kruck.com"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="some text"
              src="http://placeholder.com/articles"
            />
          </a>
        </span>
        <p
          class="author"
        >
          Bertholomule
        </p>
        <p
          class="description"
        >
          Hey friends
        </p>
      </li>
    </div>
  </div>
</DocumentFragment>
`;
